src/main/java/com/ipt/kafkatopicupdates/streams/DefaultStreamsTopology.java

```diff
@@ -0,0 +1,49 @@
+package com.ipt.kafkatopicupdates.streams;
+
+import ch.ipt.kafka.avro.Authorization;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.apache.kafka.streams.state.Stores;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
Why is this a component? Isn't this "de facto" another Utility class?

Reply:
Actually yes. I chose the `@Component` annotation to include the class in the Application Context. This enables simple DI in the `AuthenticationDeduplicationStreams.java` class. It does not need to know about the Topology. Testing is also simpler, if we want to use our own topology.